File: Ejercicios Propuestos 3/Ejercicio 8/Ejercicio8.java
import java.util.Scanner;

public class Ejercicio8 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Definir n, m, f, c Como Entero
        int n, m, f, c;

        System.out.println("Ingrese el número de filas (n):");
        n = sc.nextInt();
        System.out.println("Ingrese el número de columnas (m):");
        m = sc.nextInt();

        // Dimension Elemento[n, m], transpuesta[m, n]
        int[][] Elemento = new int[n][m];
        int[][] transpuesta = new int[m][n];

        // Elementos de la matriz
        // Para f <- 0 Hasta n - 1 Hacer
        for (f = 0; f <= n - 1; f++) {
            // Para c <- 0 Hasta m - 1 Hacer
            for (c = 0; c <= m - 1; c++) {
                System.out.println("Valor para la posición [" + f + "][" + c + "]:");
                Elemento[f][c] = sc.nextInt();
            }
        }

        // Cómo se vería la matriz original
        System.out.println("MATRIZ:");
        for (f = 0; f <= n - 1; f++) {
            for (c = 0; c <= m - 1; c++) {
                System.out.print(Elemento[f][c] + "  ");
            }
            System.out.println(""); // Salto de línea por fila
        }

        // El proceso de la matriz transpuesta
        // Para f <- 0 Hasta n - 1 Hacer
        for (f = 0; f <= n - 1; f++) {
            // Para c <- 0 Hasta m - 1 Hacer
            for (c = 0; c <= m - 1; c++) {
                transpuesta[c][f] = Elemento[f][c];
            }
        }

        // Mostrar la MATRIZ TRANSPUESTA
        System.out.println("MATRIZ TRANSPUESTA:");
        // Para f <- 0 Hasta m - 1 Hacer (Nota que ahora va hasta m - 1)
        for (f = 0; f <= m - 1; f++) {
            // Para c <- 0 Hasta n - 1 Hacer (Nota que ahora va hasta n - 1)
            for (c = 0; c <= n - 1; c++) {
                System.out.print(transpuesta[f][c] + "  ");
            }
            System.out.println(""); // Salto de línea por fila
        }

        sc.close();
    }
}

File: Ejercicios Propuestos 3/Ejercicio 8/ejercicio8.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cout << "Ingrese numero de filas: ";
    cin >> n;
    cout << "Ingrese numero de columnas: ";
    cin >> m;

    int A[50][50], T[50][50];
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cout << "A[" << i << "][" << j << "]: ";
            cin >> A[i][j];
        }
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            T[j][i] = A[i][j];
        }
    }

    cout << "Matriz transpuesta:" << endl;
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            cout << T[i][j] << " ";
        }
        cout << endl;
    }
    return 0;
}
